### PR TITLE
Add fallback buyerid for IX requests

### PR DESCRIPTION
**Pull request by muuki88**
According to index exchange this is required to get bids at all.
https://kb.indexexchange.com/publishers/openrtb_integration/list_of_supported_openrtb_bid_request_fields_for_sellers.htm#User

(cherry picked from commit 3fcaf111d1a03e81da320b955a66d52bfb8c98e1)

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/ix/IxBidder.java [hunk 1]
```diff
@@ -132,6 +136,22 @@ private ExtImpIx parseImpExt(Imp imp) {
...
     }
 
+    /**
+     * IndexExchange does nothing if `buyerid` is not present on the request.
+     * This methods adds the default `0` buyerid. Maybe IX will add the IX UID to this adapter in the future.
+     *
+     * @param user user object for the current bid request
+     * @return the user with a default value for buyerid
+     * @see [https://kb.indexexchange.com/publishers/openrtb_integration/list_of_supported_openrtb_bid_request_fields_for_sellers.htm#User]
+     */
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/ix/IxBidder.java:146]: We prefer not to use comments in bidder code

File: src/main/java/org/prebid/server/bidder/ix/IxBidder.java [hunk 2]
```diff
@@ -132,6 +136,22 @@ private ExtImpIx parseImpExt(Imp imp) {
...
+    private static User modifyUser(User user) {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/ix/IxBidder.java:147]: Put this method after modifySite method

File: src/main/java/org/prebid/server/bidder/ix/IxBidder.java [hunk 3]
```diff
@@ -132,6 +136,22 @@ private ExtImpIx parseImpExt(Imp imp) {
...
+        return user == null
+                ? User.builder().buyeruid("0").build()
+                : user.toBuilder()
+                .buyeruid(ObjectUtils.defaultIfNull(user.getBuyeruid(), "0"))
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/ix/IxBidder.java:151]: Minor. Should we use StringUtils.defaultIfEmpty()?

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/ix/IxBidder.java (hunk 1 above)
Comment (muuki88) [reply src/main/java/org/prebid/server/bidder/ix/IxBidder.java:146]: Is there a reason for that :smiley: ?

File: src/main/java/org/prebid/server/bidder/ix/IxBidder.java (hunk 3 above)
Comment (muuki88) [reply src/main/java/org/prebid/server/bidder/ix/IxBidder.java:151]: The IX docs state that an empty string is fine as well

**Comment by muuki88** [comment]
We have deployed this change and don't see any different behavior (yet). I marked this PR as draft and sent it to IX.

**Pull request closed without merging**